Guides/FirstNonNil.md

```diff
@@ -0,0 +1,43 @@
+# FirstNonNil
+
+[[Source](https://github.com/apple/swift-algorithms/blob/main/Sources/Algorithms/FirstNonNil.swift) | 
+ [Tests](https://github.com/apple/swift-algorithms/blob/main/Tests/SwiftAlgorithmsTests/FirstNonNilTests.swift)]
+
+Retrieves the first `.some` encountered while applying the given transform.
+
+This operation is available through the `firstNonNil(_:)` method on any sequence.
+
+```swift
+let value = ["A", "B", "10"].firstNonNil { Int($0) }
+// value == .some(10)
+// 
+let noValue = ["A", "B", "C"].firstNonNil { Int($0) }
+// noValue == .none
+```
+
+
+This method is analogous to `first(where:)` in how it only consumes values until
+a `.some` is found, unlike using lazy operators, which will load any sequence into a collection
```

Comment:
Could you be a little more explicit here about why this method is useful? I think @timvermeulen explained it very well in the corresponding issue, and if you get his permission, that might be a helpful piece of information for users.